samples/Suppressions.sarif

```diff
@@ -0,0 +1,130 @@
+{
+  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
+  "version": "2.1.0",
+  "properties": {
+    "comment": "This sample demonstrates how suppressions affect the visibility of results in SARIF viewers."
+  },
+  "runs": [
+    {
+      "tool": {
+        "driver": {
+          "name": "SarifSamples"
+        }
+      },
+      "results": [
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is visible because it is not suppressed."
+          }
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is hidden because it is suppressed in source."
+          },
+          "suppressions": [
+            {
+              "kind": "inSource"
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is hidden because it is suppressed externally."
+          },
+          "suppressions": [
+            {
+              "kind": "external"
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is visible because its suppression was rejected."
+          },
+          "suppressions": [
+            {
+              "kind": "external",
+              "status": "rejected"
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is hidden because its suppression was accepted."
+          },
+          "suppressions": [
+            {
+              "kind": "external",
+              "status": "accepted"
+            }
+          ]
+        },
+        {
+          "ruleId": "SMP0001",
+          "message": {
+            "text": "This result is visible because its suppression is still under review."

```

Comment:
>result is visible [](start = 26, length = 17)

disagree, we are innocent until proven guilty. that is, we should assume the suppression is good until it is rejected. otherwise, consider the user experience, they suppress the issue, open the viewer and it's _still_ visible (because it's under review). that's not right. marking it as 'review' should move it to the next person and it shouldn't be seen by the person continuing to triage results. it will bounce back as visible if rejected.

Reply:
Agreed, changed it. Also reopened [https://github.com/microsoft/sarif-web-component/issue/12](https://github.com/microsoft/sarif-web-component/issue/12) to ask Jeff to make that change to the viewer logic.

---
In reply to: [413339150](https://github.com/microsoft/sarif-tutorials/pull/12#discussion_r413339150) [](ancestors = 413339150)